nixos/modules/services/web-servers/caddy.nix

```diff
@@ -5,6 +5,23 @@ with lib;
 let
   cfg = config.services.caddy;
   configFile = pkgs.writeText "Caddyfile" cfg.config;
+
+  # v2-specific options
+  isCaddy2 = versionAtLeast cfg.package.version "2.0";
+  tlsConfig = {
+    apps.tls.automation.policies = [{
+      issuer = {
+        inherit (cfg) ca email;
+        module = "acme";
+      };
+    }];
+  };
+  adaptedConfig = importJSON (pkgs.runCommand "caddy-config-adapted.json" { } ''
+    ${cfg.package}/bin/caddy adapt --validate \
```

Comment:
Actually, thinking about it, I think `caddy adapt` should be sufficient for a syntax check, it shouldn't be additionally validated. The reason being is that validation check may complain about files not existing inside build environment.